client/blocks/hello-vote-notice/index.jsx

```diff
@@ -70,6 +72,10 @@ export default connect(
 				isSiteSection( state ) &&
 				'settings' !== getSectionName( state ) &&
 				canCurrentUser( state, selectedSiteId, 'manage_options' ) &&
+				(
+					! userStartDate ||
```

Comment:
Also, to the first point, `moment` can handle the `null` argument but would always return false:

```
> moment( null ).isBefore( moment().subtract( 1, 'weeks' ) )
false
```